eq_cir_proxy_service/utils/iap.py

```diff
@@ -0,0 +1,57 @@
+"""Utility functions for handling IAP authentication and HTTP clients."""
+
+import os
+from collections.abc import AsyncIterator
+from contextlib import asynccontextmanager
+from typing import cast
+
+import google.oauth2.id_token
+from google.auth.transport import requests
+from httpx import AsyncClient
```

Comment:
We might still want to use `httpx` and its `AsyncClient` in `main.py` too, so it is uniform across the codebase. Otherwise we will have two different approaches in the same app?